server/src/main/java/org/elasticsearch/index/codec/PerFieldMapperCodec.java

```diff
@@ -96,18 +99,25 @@ public DocValuesFormat getDocValuesFormatForField(String field) {
     }
 
     private boolean useTSDBDocValuesFormat(final String field) {
-        return isTimeSeriesModeIndex() && isNotSpecialField(field) && isNumericType(field);
+        return isTimeSeriesModeIndex() && (isCounterMetricType(field) || isTimestampField(field));
     }
 
     private boolean isTimeSeriesModeIndex() {
         return IndexMode.TIME_SERIES.equals(mapperService.getIndexSettings().getMode());
     }
 
-    private boolean isNumericType(String field) {
-        return mapperService != null && mapperService.mappingLookup().getMapper(field) instanceof NumberFieldMapper;
+    private boolean isCounterMetricType(String field) {
```

Comment:
Yes space-wise...I understand that indexing time is important but I prefer to look at the latency numbers when the benchmark runs as part of our Rally nightly benchmarks so that we can make a meaningful comparison (with similar environment). The reason why I am concentrating on space size right now is that I expect that to be a lot different even if the environment is not exactly the same.